### PR TITLE
ascanrules: Added more contextual Other Info to command OS injection rule

### PR DESCRIPTION
**Pull request by sgerlach**
Signed-off-by: Scott Gerlach <scott.gerlach@stackhawk.com>

### CODE REVIEW

**Comment by sgerlach** [comment]
Please be kind friends. We live streamed these changes :)

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CommandInjectionScanRule.java [hunk 1]
```diff
@@ -551,6 +551,11 @@ private boolean testCommandInjection(
                             .setAttack(paramValue)
                             .setEvidence(matcher.group())
                             .setMessage(msg)
+                            .setOtherInfo(
+                                    String.format(
+                                            "The scanner was able to retrieve the content of a file or command "
```
Comment (thc202) [inline addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CommandInjectionScanRule.java:556]: - These should be moved to `Messages.properties` file.
 - Be good to assert these in the tests.

**Comment by kingthorin** [comment]
Is there a recording?

Thanks for tackling that.

**Comment by sgerlach** [comment]
Of course there is https://youtu.be/X59Hbonqm3E

File: addOns/ascanrules/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
 ### Changed 
+- Command injection now has more information in the Other Info field to differentiate feedback or blind tests
```
[the remark this answers is not in the document]
Comment (sgerlach) [reply addOns/ascanrules/CHANGELOG.md]: the two types of tests feedback-based or time-based

**Comment by kingthorin** [comment]
Nice work. That’s rough about the audio but oh well you put yourself out there and did it live, that’s great!

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CommandInjectionScanRule.java (hunk 1 above)
Comment (sgerlach) [reply addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CommandInjectionScanRule.java]: Tried really hard here.... Let me know if it's OK (I couldn't figure out the [0] based interpolation)

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CommandInjectionScanRule.java [hunk 3]
```diff
@@ -350,6 +350,11 @@ public int getRisk() {
         return Alert.RISK_HIGH;
     }
 
+    private String getOtherInfo(String testType, String testValue) {
+        return String.format(
+                Constant.messages.getString(MESSAGE_PREFIX + "otherinfo." + testType), testValue);
+    }
+
```
Comment (kingthorin) [inline addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CommandInjectionScanRule.java:357]:
```suggestion
    private String getOtherInfo(String testType, String testValue) {
        return Constant.messages.getString(MESSAGE_PREFIX + "otherinfo." + testType), testValue);
    }

```

File: addOns/ascanrules/src/main/resources/org/zaproxy/zap/extension/ascanrules/resources/Messages.properties [hunk 4]
```diff
@@ -9,6 +9,8 @@ ascanrules.bufferoverflow.other = Potential Buffer Overflow.  The script closed
 ascanrules.commandinjection.name=Remote OS Command Injection
 ascanrules.commandinjection.desc=Attack technique used for unauthorized execution of operating system commands. This attack is possible when an application accepts untrusted input to build operating system commands in an insecure manner involving improper data sanitization, and/or improper calling of external programs.
 ascanrules.commandinjection.refs=http://cwe.mitre.org/data/definitions/78.html\nhttps://owasp.org/www-community/attacks/Command_Injection
+ascanrules.commandinjection.otherinfo.feedback-based=The scanner was able to retrieve the content of a file or command by sending '/%s' to the operating system running this application
+ascanrules.commandinjection.otherinfo.time-based=The scanner was able to control the timing of the application response by sending '/%s' to the operating system running this application
```
Comment (kingthorin) [inline addOns/ascanrules/src/main/resources/org/zaproxy/zap/extension/ascanrules/resources/Messages.properties:13]:
```suggestion
ascanrules.commandinjection.otherinfo.feedback-based=The scanner was able to retrieve the content of a file or command by sending {0} to the operating system running this application
ascanrules.commandinjection.otherinfo.time-based=The scanner was able to control the timing of the application response by sending {0} to the operating system running this application
```

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CommandInjectionScanRule.java (hunk 3 above)
Comment (kingthorin) [reply addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CommandInjectionScanRule.java:357]: Probably needs a spotlessApply after that.

Comment (thc202) [reply addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CommandInjectionScanRule.java:357]: `testType),` → `testType,`

Comment (sgerlach) [reply addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CommandInjectionScanRule.java:357]: AHHHHHHHH! Thanks!

**Review by kingthorin — APPROVED** [review]

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CommandInjectionScanRule.java (hunk 3 above)
Comment (kingthorin) [reply addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CommandInjectionScanRule.java]: oops, thanks @thc202

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**